lib/std/Thread.zig

```diff
@@ -76,6 +76,34 @@ pub fn spinLoopHint() void {
     }
 }
 
+pub fn setName(self: Thread, name: [:0]const u8) !void {
+    if (use_pthreads) {
+        const err = c.pthread_setname_np(self.data.handle, name.ptr);
+        return switch (err) {
+            0 => {},
+            os.ERANGE => error.NameTooLong,
+            else => unreachable,
+        };
+    }
+}
+
+pub fn getName(self: Thread, allocator: *std.mem.Allocator) !?[:0]const u8 {
+    if (use_pthreads) {
+        // The thread name cannot be longer than 16 bytes including the null terminator.
```

Comment:
> The thread name  is  a  meaningful  C  language
       string, whose length is restricted to 16 characters, including the ter‐
       minating null byte ('\0').

Is this limitation lifted for other platforms?

Reply:
I don't think so, I wrote this comment before realising `pthread_setname_np` is non standard and looking at the man pages of the BSDs they don't mention a length limitation. 

I'll change the comment.

Reply:
FYI, OpenBSD has 32 bytes limit ('\0' included). the string is copied and silenciously truncated to fit